build_tools/circle/build_wheels_arm.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+set -e
+set -x
+
+sudo add-apt-repository --remove ppa:ubuntu-toolchain-r/test
+sudo apt-get update
+sudo apt-get install python3-virtualenv ccache
+python3 -m virtualenv --system-site-packages --python=python3 testenv
+source testenv/bin/activate
+pip install --upgrade pip
+for pyversion in 37 38 39; do
```

Comment:
I am afraid that building the ARM wheels sequentially will slow down the release process. Actually, for 3 python, circle ci seems to be able to complete in 25m which is not too bad. Maybe this is because of ccache?

Still we might want to have one CI job per Python version and leverage Circle CI parallelism https://circleci.com/docs/2.0/parallelism-faster-jobs/. Although one would have to check how many concurrent workers are offered by default in the free Circle Ci account.

Reply:
Actually I am confused when reading the logs of the cibuildwheel: it seems that it's always using Python3.8 to build all the 3 wheel versions. This means that we will build 3.7 wheels against the Python 3.8 ABI.

I think we need to make sure that each Python version is run in its own circleci job config within an image with the matching Python version: https://circleci.com/docs/2.0/circleci-images/#python (3.7, 3.8 and 3.9).

Reply:
Unfortunately, according to https://circleci.com/pricing/ I don't think we can benefit from parallelism. So no need to try to get this to work.

However we still need to split the for loop into independent jobs to make sure we use the right Python version for each generated wheel.

Reply:
> I think we need to make sure that each Python version is run in its own circleci job config within an image with the matching Python version: https://circleci.com/docs/2.0/circleci-images/#python (3.7, 3.8 and 3.9).

If I understand correctly those customized python images are not available for arm (see the [available images for arm documentation](https://circleci.com/docs/2.0/arm-resources/#overview)), this means to reinstall a conda environment for each wheel on the host image `ubuntu-2004:202101-01`. 

> Actually I am confused when reading the logs of the cibuildwheel: it seems that it's always using Python3.8 to build all the 3 wheel versions. This means that we will build 3.7 wheels against the Python 3.8 ABI.

I am confused too then... :) ... I was thinking that the `CIBW_BUILD` variable was requesting a specific python version in the `manylinux` image... eg [this line](https://app.circleci.com/pipelines/github/scikit-learn/scikit-learn/16867/workflows/66cbe705-61df-4d38-b6f4-593ab0623498/jobs/148909?invite=true#step-102-551) shows that python 3.9 is called ... but it is not clear to me if it is during the build of the wheel... perhaps I should also add a bit of verbosity during the WIP state.

Anyway, I understand your concern, let me create the conda envs then.

Reply:
Yes you are right! For some reason I thought cibuildwheel used docker only for the repair step but it's indeed actually also used for the build step: https://github.com/pypa/cibuildwheel/blob/144733dc60fe0c9c3114e7ef2e4996ad919fec5c/cibuildwheel/linux.py#L186

Too bad that the `which python` before in the log does not print its output in the log, that would have lifted the ambiguity.

Reply:
I have configured `CIBW_BUILD_VERBOSITY: 1` as for the other wheels. Hope this will help.